File: db-query-backend/src/nl/prompt.builder.ts
type TableSchema = {
  name: string;
  columns: string[];
};

export function buildNLtoSQLPrompt(
  dbType: string,
  schema: { name: string; columns: string[] }[],
  question: string,
) {
  const schemaText = schema
    .map(t => `${t.name}: ${t.columns.join(', ')}`)
    .join('; ');

  return `
Generate ONE ${dbType} SQL SELECT query.

Schema:
${schemaText}

Rules:
- Use only listed tables & columns
- No explanations
- Return SQL only

Question:
${question}
`;
}

File: db-query-backend/app/llm/prompt_builder.py
"""Port of nl/prompt.builder.ts — identical prompt text, used by all three providers."""
from app.models.dto import TableSchema


def build_nl_to_sql_prompt(db_type: str, schema: list[TableSchema], question: str) -> str:
    schema_text = "\n".join(f"{t.name}: {', '.join(t.columns)}" for t in schema)
    return f"""Generate ONE {db_type} SQL SELECT query.

Schema:
{schema_text}

Rules:
- Use only listed tables & columns
- No explanations
- Return SQL only

Question:
{question}"""
